### DISCUSSION TITLE
cmakeproject() That reads CMakeLists.txt and automatically creates a premake project.

### DISCUSSION THREAD

**Discussion by babaliaris** [premake/premake-core#1830] [Ideas]
It would be awesome to do something like the following:

```lua
workspace "MyWorkSpace"
    ...
    cmakeproject "ExternalProject"
        location "directory/where/CMakeLists/is/located/"
        kind "StaticLib"
        ...

    project "MyApp"
        kind "ConsoleApp"
        links "ExternalProjec"
        ...
    ...
```

And this would automatically read the CMakelist and convert it to a Premake project, no matter the version of CMake.

I have no idea how Premake or Cmake works under the hood, but I guess because CMake has lots of versions, that would be an insane workload to do?

But imagine the power of Premake if this could be achieved. Not to mention that will increase its popularity!
Tones of c/c++ projects out there are using CMake, and currently, it's hard to work with both CMake and Premake.
You have to do a lot of CMake generation and compilation yourself first and include the correct **libdirs** and **links** manually.

**Comment by Jarod42** [comment]
There is [premake-cmake](https://github.com/Enhex/premake-cmake) (and my [fork](https://github.com/Jarod42/premake-cmake)) which generates CMake from premake.
I'm not aware of CMake to premake version though.
